src/rgw/rgw_crypt.cc

```diff
@@ -1214,11 +1343,13 @@ int rgw_s3_prepare_decrypt(struct req_state* s,
     res = get_actual_key_from_kms(s->cct, key_id, key_selector, actual_key);
     if (res != 0) {
       ldout(s->cct, 10) << "No encryption key for key-id=" << key_id << dendl;
+      s->err.message = key_id + " does not exist";
```

Comment:
this could fail for other reasons, like a misconfigured kms or network issues. how about `Failed to retrieve the kms-keyid` here?

Reply:
OK, I have changed it to `s->err.message = "Failed to retrieve the actual key, kms-keyid: " + key_id;`